packages/vehicle-rental-overlay/src/index.js

```diff
@@ -0,0 +1,251 @@
+import { divIcon } from "leaflet";
+import memoize from "lodash.memoize";
+import * as BaseMapStyled from "@opentripplanner/base-map/lib/styled";
+import { getCompaniesLabelFromNetworks } from "@opentripplanner/core-utils/lib/itinerary";
+import {
+  companyType,
+  mapSymbolsType,
+  stationsType
+} from "@opentripplanner/core-utils/lib/types";
+import FromToLocationPicker from "@opentripplanner/from-to-location-picker";
+import PropTypes from "prop-types";
+import React from "react";
+import ReactDOMServer from "react-dom/server";
+import {
+  CircleMarker,
+  FeatureGroup,
+  Marker,
+  MapLayer,
+  Popup,
+  withLeaflet
+} from "react-leaflet";
+
+import { floatingBikeIcon, hubIcons } from "./bike-icons";
+import * as Styled from "./styled";
+
+const getStationMarkerByColor = memoize(color =>
+  divIcon({
+    className: "",
+    iconSize: [11, 16],
+    popupAnchor: [0, -6],
+    html: ReactDOMServer.renderToStaticMarkup(
+      <Styled.StationMarker color={color} />
+    )
+  })
+);
+
+class VehicleRentalOverlay extends MapLayer {
+  createLeafletElement() {}
+
+  updateLeafletElement() {}
+
+  startRefreshing() {
+    // ititial station retrieval
+    this.props.refreshVehicles();
+
+    // set up timer to refresh stations periodically
+    this.refreshTimer = setInterval(() => {
+      this.props.refreshVehicles();
+    }, 30000); // defaults to every 30 sec. TODO: make this configurable?*/
+  }
+
+  stopRefreshing() {
+    if (this.refreshTimer) clearInterval(this.refreshTimer);
+  }
+
+  componentDidMount() {
+    const { companies, mapSymbols, name, visible } = this.props;
+    if (visible) this.startRefreshing();
+    if (!mapSymbols)
+      console.warn(`No map symbols provided for layer ${name}`, companies);
+  }
+
+  componentWillUnmount() {
+    this.stopRefreshing();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!prevProps.visible && this.props.visible) {
+      this.startRefreshing();
+    } else if (prevProps.visible && !this.props.visible) {
+      this.stopRefreshing();
+    }
+  }
+
+  /**
+   * Render some popup html for a station. This contains custom logic for
+   * displaying rental vehicles in the TriMet MOD website that might not be
+   * applicable to other regions.
+   */
+  renderPopupForStation = (station, stationIsHub = false) => {
+    const { configCompanies, setLocation } = this.props;
+    const stationNetworks = getCompaniesLabelFromNetworks(
+      station.networks,
+      configCompanies
+    );
+    let stationName = station.name || station.id;
+    if (station.isFloatingBike) {
+      stationName = `Free-floating bike: ${stationName}`;
+    } else if (station.isFloatingCar) {
+      stationName = `${stationNetworks} ${stationName}`;
+    } else if (station.isFloatingVehicle) {
+      // assumes that all floating vehicles are E-scooters
+      stationName = `${stationNetworks} E-scooter`;
```

Comment:
Since we're touching this code, it might be worth extracting the `E-scooter` string into a prop that can be overridden if need be.